packages/cdktf/lib/terraform-asset.ts

```diff
@@ -0,0 +1,102 @@
+import { Construct, ISynthesisSession, Node } from "constructs";
+import * as fs from "fs";
+import * as path from "path";
+import { Manifest } from "./manifest";
+import { copySync, hashPath } from "./private/fs";
+import { Resource } from "./resource";
+
+export interface TerraformAssetConfig {
+  readonly path: string;
+  readonly type?: AssetType;
+  readonly assetHash?: string;
+}
+
+export enum AssetType {
+  FILE,
+  DIRECTORY,
+}
+
+const ASSETS_DIRECTORY = "assets";
+
+export class TerraformAsset extends Resource {
+  private sourcePath: string;
+  public assetHash: string;
+  public type: AssetType;
+
+  constructor(scope: Construct, id: string, config: TerraformAssetConfig) {
+    super(scope, id);
+
+    if (!path.isAbsolute(config.path)) {
+      throw new Error(
+        `TerraformAsset path needs to be absolute, got relative path: '${config.path}'`
+      );
+    }
+
+    const stat = fs.statSync(config.path);
+    const inferredType = stat.isFile() ? AssetType.FILE : AssetType.DIRECTORY;
+    this.type = !config.type ? inferredType : config.type;
```

Comment:
Minor: any reason not to use `this.type = config.type ?? inferredType`?

Reply:
No, I'll change it